generators/server/prompts.js

```diff
@@ -101,6 +101,31 @@ function askForServerSideOpts() {
         },
         {
             when: function (response) {
+                return applicationType === 'gateway' || applicationType === 'microservice' || applicationType === 'uaa';
```

Comment:
if you choose uaa for microservices, the prompt is first asking "uaa?", then "discovery" and then some uaa related question to find the uaa for uaaBasename....this is confusing, please move the registry question after the uaa, so it asks me for uaa directory before service discovery